debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -186,7 +195,7 @@ public Instant timestampOfLsn(Lsn lsn) throws SQLException {
             statement.setBytes(1, lsn.getBinary());
         }, singleResultMapper(rs -> {
             final Timestamp ts = rs.getTimestamp(1);
-            final Instant ret = (ts == null) ? null : ts.toInstant();
```

Comment:
I'm trying to wrap my head around this; in your description above you said:

 > Currently, the server timestamp extracted by SqlServerConnection.timestampOfLsn() is a DateTime and don't own any timezone information.

But that seems not to be the case here. `Timestamp` extends `java.util.Date` and as such represents a specific instant in time (as does `Instant`), so I still don't quite understand where the TZ drift should be coming from.

Reply:
From my understanding :
* Debezium use `sys.fn_cdc_map_lsn_to_time` to extract the transaction timestamp
* This function itself get the transaction end date from `trans_end_time` column of the `[cdc].[lsn_time_mapping]` table
* `trans_end_time` is a `DateTime`
* `DateTime` don't have timezone (https://docs.microsoft.com/en-us/sql/t-sql/functions/date-and-time-data-types-and-functions-transact-sql?view=sql-server-ver15)
* And because java resultset provide a getTimestamp function, it do what it can with a DateTime. It means that it has no other timezone available to use that the client timezone and then do a wrong assumption about the date timezone 
* Probably `trans_end_time` should be a `datetimeoffset`... But i'm not so sure I want to patch sql server ^_^

Reply:
Thanks, got it now, I observe the behaviour you describe.

That being said, I came across [this page](https://docs.microsoft.com/en-us/sql/t-sql/functions/current-timezone-transact-sql?view=sql-server-ver15): Seems we could simply call `SELECT CURRENT_TIMEZONE ( )` (and it might always be UTC anways)?

Reply:
There's also `SELECT SYSDATETIMEOFFSET()` which returns the offset of the DB's TZ in comparison to UTC. Either one should allow us to properly shift the timezone. WDYT?

Reply:
@gunnarmorling I've already proposed that yesterday - see link https://docs.microsoft.com/en-us/sql/t-sql/functions/current-timezone-transact-sql?view=sql-server-ver15
`SELECT SYSDATETIMEOFFSET()` also looks good but we need to verify which of those two functions take system TZ into consideration and which one database TZ

Reply:
Actually `SELECT CURRENT_TIMEZONE ( )` doesn't work on my sql server (Microsoft SQL Server 2016 (SP2-CU7) (KB4495256) - 13.0.5337.0 (X64))
but SYSDATETIMEOFFSET do the job. 
Here is the kind of info we can extract with it :
 ```sql
SELECT
    SYSDATETIMEOFFSET() AS [System DateTime Offset],
    DATEPART(TZoffset, SYSDATETIMEOFFSET()) AS [Timezone Offset],
        FORMAT(SYSDATETIMEOFFSET(), 'zz') AS 'zz',
    FORMAT(SYSDATETIMEOFFSET(), 'zzz') AS 'zzz';
```
But compared to the xp_regread approach maybe dealing with winter/summer time shift could be tricky. We have to think about that...

Reply:
> dealing with winter/summer time shift is not possible.

Is this a problem actually? You'd get the TZ offset from the database and you can get the current TZ offset (which may or may not include summer time, it doesn't matter though, all you care about is the offset value, e.g. `02:00`) of the JVM. Based on the difference between the two you can adjust the `TIMESTAMP` accordingly and get the right `Instant`.

Reply:
Not totally.Imagine with a time shift from GMT+1 to GMT+2  the 29 March at 2:00AM (CET). 
* you udpate your database at 2020-03-29 (1:45AM GMT+1) (0:45 UTC)
  transaction end datetime = 1:45AM (no offset or zone stored)
* you ingest the change in kafka after the shift through debezium
* get the transaction date 1:45AM (no offset or zone stored)
* get the offset (SELECT SYSDATETIMEOFFSET()): --> GMT+2
* so UTC transaction date become 23:44

Compared to Offset, Timezone contains the metadata about what is the time shift for a given instant
So if you have a datetime you can know the timeshift for the given region/timezone. winter, summer no matter.
Offset will also force us to periodically/systematically ask for the server offset it case their was a shift compared to the last run

Reply:
I would personally suggest to switch the entire universe to UTC. It would so much easier !
That's my plan B
plan C is the sql server patch

Reply:
Agreed, I'd like that plan A :)